### ISSUE TITLE
Gatsby using deprecated @hapi/joi

### ISSUE THREAD

**Issue by hyecheol123** [gatsbyjs/gatsby#25764]
## Description

Gatsby is depending on the deprecated dependencies

### Steps to reproduce

Install Command as guide in official document
- `npm install -g gatsby-cli`

### Expected result

Expecting no warnings for deprecated packages, and expected to see clean output with success message

### Actual result

```
npm WARN deprecated @hapi/joi@15.1.1: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/address@2.1.4: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
```

### Environment

  System:
    OS: Linux 3.10 Debian GNU/Linux 10 (buster) 10 (buster)
    CPU: (8) arm64 unknown
    Shell: 5.0.3 - /bin/bash
  Binaries:
    Node: 14.5.0 - ~/.nvm/versions/node/v14.5.0/bin/node
    npm: 6.14.5 - ~/.nvm/versions/node/v14.5.0/bin/npm
  npmGlobalPackages:
    gatsby-cli: 2.12.60

### Other Observation
It also complains that the dependency for acorn-dynamic-import is missing (acorn@^6.0.0).
[ERR MSG]
```
npm WARN acorn-dynamic-import@4.0.0 requires a peer of acorn@^6.0.0 but none is installed. You must install peer dependencies yourself.
```

**Comment by ascorbic** [comment]
Thanks for the report. This is because `gatsby` and `gatsby-recipes` are using an old version of [@hapi/joi](https://github.com/sideway/joi) for node validation. The newer version is a breaking change, though I don't know if it breaks our schema specifically. If someone wants to take this on, they'd need to look at the changelog for Joi and see if it affects us.

**Comment by herecydev** [comment]
I'll take a look at this, the changelogs are **insanely long** so I'll see if it builds first and then go from there

**Comment by ascorbic** [comment]
Thanks, @herecydev. Much appreciated. If it passes tests, the main issue will be checking for false passes with invalid input.

**Comment by herecydev** [comment]
Sure I'll see what tests are there and potentially right some further invalid tests

**Comment by iAbdullahWaleed** [comment]
Hey Guys, This problem prevent me from downloading gatsby, so there is any solution or I should wait?

**Comment by mxstbr** [comment]
@iAbdullahWaleed deprecated dependencies are warnings when `npm install`ing, not errors, so you should be able to use gatsby anyway? :blush:

**Comment by iAbdullahWaleed** [comment]
Yes I can, but it's was my first time using the command line, so i was confused when i saw the warnings 🙄.

**Comment by mxstbr** [comment]
Makes a lot of sense, they are really confusing! Hopefully we can get rid of them soon, until then enjoy Gatsby!

**Comment by hyecheol123** [comment]
Looks like it is hard to fix this issue, but thanks for all of your hard works and efforts to solve this problem!!
Sorry that I am not able to help the process, as I have no knowledge on Javascript.

**Comment by PluckyPrecious** [comment]
Hi @mxstbr it's my first time installing Gatsby too,  but after getting the deprecated dependencies warning an error occur without complete installation on windows 10. 

```
npm ERR! Unexpected end of JSON input while parsing near '...+oQAJqigTCokVza05RpiQ'
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\HP\AppData\Roaming\npm-cache\_logs\2020-07-30T04_28_22_170Z-debug.log
```

**Comment by aminocode0** [comment]
I'm also facing the same issue in my **Linux** system, please help me to get out from this gatsby error. please help me, anyone

**Comment by herecydev** [comment]
There's a pr here: https://github.com/gatsbyjs/gatsby/pull/25797

The fix is in progress but it's moving slowly.

**Comment by pavan-elisetty** [comment]
same issue , though @mxstbr said these are warnings but we will be able to use gatsby , but i am unable to use gatsby cli 
![image](https://user-images.githubusercontent.com/57026704/93764714-391cb080-fc31-11ea-88cd-7da2014f6ab4.png)

**Comment by adriatic** [comment]
This issue is still unresolved (I tripped on it today in a different context, running ` npm install react-router-dom`). Does this mean that that it is the responsibility of the team that maintains the `react-router-dom` module, or the Gatsby team od Strapi team? Either way why is the process of addressing this issue taking so long?

**Comment by herecydev** [comment]
@adriatic See this comment: https://github.com/gatsbyjs/gatsby/pull/25797#issuecomment-669392219

If you can help migrate that file; that will unblock this PR.

**Comment by adriatic** [comment]
I am torn, Dan (@herecydev) by your proposal; encountered this problem doing something very different, so from that point of view the remaining work on **[this issue](https://github.com/gatsbyjs/gatsby/pull/25797#issuecomment-669392219)** is a very low priority for me. On the other hand, probably all people encountering it think the same 😄.

**Comment by Code-Tap** [comment]
This still appears to be an issue.
Clean first time install

![2020-12-17-063754_1266x344_scrot](https://user-images.githubusercontent.com/23399537/102452609-86db6600-4032-11eb-8829-16ac472c3051.png)

**Comment by yogoeasy** [comment]
I failed to install Gatsby CLI with npm. Instead, I use yarn to install it successfully.

**Comment by mihanizm56** [comment]
can not build my project 
    "gatsby": "4.1.0",

default cli configuration
    
    node 14.18.1
    
    
![image](https://user-images.githubusercontent.com/37120827/140545068-1b3741ab-ad5a-4d76-8cee-988548c10878.png)

**Comment by marvinjude** [comment]
Closing this because it has been addressed in https://github.com/gatsbyjs/gatsby/pull/29792
